script/testing/oltpbench/README.md

```diff
@@ -0,0 +1,65 @@
+# OLTP Benchmark Testing
+This folder include the necessary files for running oltpbench testings.
+
+# OLTP benchmark types
+Currently we run the following OLTP benchmarks
+- TATP
+- TPCC
+- NOOP
+- YCSB
+- Smallbank
+
+## External dependencies
+- [oltpbench](https://github.com/oltpbenchmark/oltpbench): for executing the OLTP benchmark of your choice
+
+## Internal dependencies
+Let the base directory be `noisepage/script/testing`
+- `util/`: provides base classes/functions/constants to construct the `TestOLTPBench` class
+  - `test_server.py`
+    - Contains the base `TestServer` class to manage the lifecycle of a test suite for a certain OLTP benchmark type
+  - `test_case.py`
+    - Contains the base `TestCase` class to manage the lifecycle of a test case
+  - `common.py`
+    - Contains common utility helper functions
+  - `constants.py`
+    - Contains general constants, like `ErrorCode`
+- `reporting/`: provide functions to report testing results to Django API
+
+## File structures
+Let the base directory be `noisepage/script/testing/oltpbench`
+- `configs/`: contains the JSON config files for Jenkins to run OLTPBench tests for 
+  - End-to-end debugging
+  - End-to-end performance
+  - Nightly
+
+  For more details, please refer to [Configuration file for OLTPBench](https://github.com/cmu-db/noisepage/blob/master/script/testing/oltpbench/configs/README.md)
+- `run_oltpbench.py`: entry point to run the OLTP bench tests
+- `test_oltpbench.py`: defines the `TestOLTPBench` class which manage the lifecycle of a test suite
+- `test_case_oltp.py`: defines the `TestOLTPBench` class which manage the lifecycle of a test case
+- `utils.py`: defines a list of utility functions specifically used by OLTP bench tests
+
+## Test workflow
+- Start the NoisePage DB process as a Python subprocess
```

Comment:
I think in this section it's important to say what each component does. We could also abstract into sections if that makes it easier. For example.
```
run_oltpbench.py
- will load the config file passed in and start an instance of the database with any server_args specified in the config. 
- It will kill any lingering processes
- It will create a test suite for the whole config file and runs the test suite
TestOLTPBench
- downloads and installs and builds OLTPBench from github
- Then it runs all the TestCases specified in the config (more details)

OLTPBenchTestCase
-...
-...


```

Reply:
Wait, do you mean who-does-what for each of the test stage?

Reply:
Done